src/en/mangasaki/src/eu/kanade/tachiyomi/extension/en/mangasaki/MangaSaki.kt

```diff
@@ -0,0 +1,265 @@
+        } else {
+            // The site doesn't show thumbnails when using search
+            val titleElement = element.select("h3.title a")
+            Manga.title = titleElement.text()
+            Manga.setUrlWithoutDomain(titleElement.attr("href"))
+        }
+
+        return Manga
+    }
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    // manga details
+    override fun mangaDetailsParse(document: Document): SManga {
+        val Manga = SManga.create()
+        Manga.author = document.select("div.field-name-field-author div.field-item").first()?.text()
+        Manga.genre = document.select("div.field-name-field-genres ul li a").joinToString { it.text() }
+        Manga.description = document.select("div.field-name-body div.field-item p").text()
+        Manga.thumbnail_url = document.select("div.field-name-field-image2 div.field-item img").attr("src")
+
+        document.select("div.field-name-field-status div.field-item").text().also { statusText ->
+            when {
+                statusText.contains("Ongoing", true) -> Manga.status = SManga.ONGOING
+                statusText.contains("Complete", true) -> Manga.status = SManga.COMPLETED
+                else -> Manga.status = SManga.UNKNOWN
+            }
+        }
+
+        return Manga
+    }
+
+    // chapters
+    override fun chapterListRequest(manga: SManga) = chapterListRequest(manga.url, 1)
+
+    private fun chapterListRequest(url: String, page: Int): Request {
+        return GET("$baseUrl$url?page=${page - 1}", headers)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        var document = response.asJsoup()
+        val chapters = document.select(chapterListSelector()).map(::chapterFromElement).toMutableList()
+        var nextPage = 2
+
+        while (document.select(paginationNextPageSelector).isNotEmpty()) {
+            val dirtyPage = document.select("div#block-search-form form#search-block-form").attr("action")
+            val cleaningIndex = dirtyPage.lastIndexOf("?")
+            val cleanPage = dirtyPage.substring(0, cleaningIndex)
+            document = client.newCall(chapterListRequest(cleanPage, nextPage)).execute().asJsoup()
+            chapters += document.select(chapterListSelector()).map(::chapterFromElement)
+            nextPage++
+        }
+
+        return chapters
+    }
+
+    private val paginationNextPageSelector = latestUpdatesNextPageSelector()
+
+    override fun chapterListSelector() = ".chlist tbody tr"
+
+    override fun chapterFromElement(element: Element): SChapter {
+        val Chapter = SChapter.create()
+        Chapter.setUrlWithoutDomain(element.select("a").attr("href"))
+        Chapter.name = element.select("a").text()
+        Chapter.date_upload = element.select("td").last()?.text()?.let {
+            SimpleDateFormat("dd MMM yyyy", Locale.ENGLISH).parse(it)?.time ?: 0
+        } ?: 0
+
+        return Chapter
+    }
```

Comment:
```suggestion
    private val dateFormat = SimpleDateFormat("dd MMM yyyy", Locale.ENGLISH)

    override fun chapterFromElement(element: Element): SChapter {
        val chapter = SChapter.create()
        chapter.setUrlWithoutDomain(element.select("a").attr("href"))
        chapter.name = element.select("a").text()
        chapter.date_upload = element.select("td").last()?.text()?.let {
            dateFormat.parse(it)?.time ?: 0
        } ?: 0

        return chapter
    }
```
Reuse the `SimpleDateFormat` object instead of creating a new one every chapter.